R/zchunk_LA143.HDDCDD.R

```diff
@@ -31,65 +30,184 @@ module_energy_LA143.HDDCDD_DISABLED <- function(command, ...) {
+        year = substr(year,2,5),
+        # Assuming that the variable is the 3rd word in the file name (separated by "_")
+        variable = substr( file,
+                regexpr( "DD_", file, fixed = T ) - 1,
+                regexpr( "DD_", file, fixed = T ) + 1 ),
+        # Assuming that the GCM comes after "DD_" and is 6 letters
+        GCM = substr( file,
+                regexpr( "DD_", file, fixed = T ) + 3,
+                regexpr( "DD_", file, fixed = T ) + 8 ),
+        # Assuming that the last word is the scenario, starting ten letters pass DD_
+        SRES = substr( file,
+                regexpr( "DD_", file, fixed = T ) + 10,
+                length(file)),
+        # Set all negative values to 0
+        value = if_else(value < 0, 0, value)
+      )
+
+    if(OLD_DATA_SYSTEM_BEHAVIOR){
+    # Add in country iso
+    L143.HDDCDD_scen_ctry_Y <- HDDCDD_data %>%
+      # Drop file name
+      select(-file) %>%
+      # Filter only useful years
+      filter( year %in% c(HISTORICAL_YEARS, FUTURE_YEARS)) %>%
+      mutate(year = as.integer(year)) %>%
+      # Drop Cote d'Ivoire--this is a mistake in old data system
+      filter(country != "Cote d'Ivoire") %>%
+      left_join_error_no_match(GIS_ctry, by = 'country')
+    } else{
+      # Add in country iso
+      L143.HDDCDD_scen_ctry_Y <- HDDCDD_data %>%
+        # Drop file name
+        select(-file) %>%
+        # Filter only useful years
+        filter( year %in% c(HISTORICAL_YEARS, FUTURE_YEARS)) %>%
+        mutate(year = as.integer(year)) %>%
+        # Remove apostrophe in Cote d'Ivoire and add in country iso by country name
+        mutate(country = if_else(country == "Cote d'Ivoire", "Cote dIvoire", country)) %>%
+        left_join_error_no_match(GIS_ctry, by = 'country')
+    }
+
+    # Serbia and Montenegro are currently combined. Copy to separated countries, assigning the same HDD and CDD to each
+    if( "scg" %in% L143.HDDCDD_scen_ctry_Y$iso ){
+      # Create Serbia tibble
+      L143.HDDCDD_scen_srb_Y <- L143.HDDCDD_scen_ctry_Y %>%
+        filter(iso == "scg" ) %>%
+        mutate(iso = "srb")
+
+      # Insert Serbia tibble, change 'scg' iso to 'mne' iso
+      L143.HDDCDD_scen_ctry_Y <- L143.HDDCDD_scen_ctry_Y %>%
+        bind_rows(L143.HDDCDD_scen_srb_Y) %>%
+        mutate(iso = if_else(iso == "scg", "mne", iso))
+    }
+
+    # Add population data and region data
+    L143.wtHDDCDD_scen_ctry_Y <- L143.HDDCDD_scen_ctry_Y %>%
+      # Join with population data converted to long format
+      left_join_error_no_match(L101.Pop_thous_GCAM3_ctry_Y,
+                               by = c('iso', 'year')) %>%
+      # Join with region ID data
+      left_join_error_no_match(iso_GCAM_regID, by = 'iso')
+
+    # Old behavior divides total population*DD by total population in region, but
+    # total population in region includes countries that don't have degree days recorded.
+    # New behavior finds weighted mean with DD as values and population as weights.
```

Comment:
Yes, the old behavior was a mistake. For example, in region 1, the Virgin Islands are included in the population file but not the degree day file. The degree days were then calculated as:
(DD_usa*pop_usa + DD_puertorico* pop_puertorico)/(pop_usa + pop_puertorico + pop_virginislands)